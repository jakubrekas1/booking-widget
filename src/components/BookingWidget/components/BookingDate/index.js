export { default as BookingDate } from './BookingDateContainer';
