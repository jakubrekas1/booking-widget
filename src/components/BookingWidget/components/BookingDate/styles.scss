@import '../../../../assets/styles/colors';

.booking-date{
  margin-bottom: 25px;

  &__label{
    margin-bottom: 6px;
    color: $black;
    font-size: 14px;
    font-weight: bold;
  }

  &__wrapper{
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 8px;
    border: 1px solid darken($white, 10%);
    border-radius: 2px;
  }

  &__arrow{
    width: 30px;
    height: 30px;
    padding: 0 30px 0 10px;
    background-image: url('../../../../assets/images/arrow-right.svg');
    background-position: center;
    background-size: contain;
    background-repeat: no-repeat;
  }
}
