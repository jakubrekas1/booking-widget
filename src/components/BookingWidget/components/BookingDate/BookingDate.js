import React from 'react';
import PropTypes from 'prop-types';

import { Input } from './components/Input';

const BookingDate = ({
  dateEnd,
  dateStart,
  onClick,
  showCalendar,
}) => {
  const onEndDateClick = () => onClick(dateEnd);
  const onStartDateClick = () => onClick(dateStart);

  return (
    <div className="booking-date__outer">
      {showCalendar && <div>Datepicker here</div>}
      <div className="booking-date">
        <p className="booking-date__label">Dates</p>
        <div className="booking-date__wrapper">
          <Input onClick={onStartDateClick} placeholder="Check In" value={dateStart} />
          <img
            alt="Arrow"
            className="booking-date__arrow"
            src="../../../../assets/images/arrow-right.svg"
          />
          <Input onClick={onEndDateClick} placeholder="Check Out" value={dateEnd} />
        </div>
      </div>
    </div>
  );
};

BookingDate.defaultProps = {
  dateEnd: '',
  dateStart: '',
  showCalendar: false,
};

BookingDate.propTypes = {
  dateEnd: PropTypes.string,
  dateStart: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  showCalendar: PropTypes.bool,
};

export default BookingDate;
