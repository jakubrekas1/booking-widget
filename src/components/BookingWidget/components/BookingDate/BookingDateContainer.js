import React, { useState } from 'react';

import BookingDate from './BookingDate';

const BookingDateContainer = () => {
  const [defaultCurrentDate, setDefaultCurrentDate] = useState(null);

  const [dateStart, setDateStart] = useState('');
  const [dateEnd, setDateEnd] = useState('');

  const [showCalendar, setShowCalendar] = useState(false);

  const handleCalendarChange = (range) => {
    setDateStart(range.dateStart);
    setDateEnd(range.dateEnd);
  };

  const handleDateClick = (date) => {
    setDefaultCurrentDate(date);
    setShowCalendar(true);
  };

  return (
    <BookingDate
      dateEnd={dateEnd}
      dateStart={dateStart}
      defaultCurrentDate={defaultCurrentDate}
      onCalendarChange={handleCalendarChange}
      onClick={handleDateClick}
      showCalendar={showCalendar}
    />
  );
};

export default BookingDateContainer;
