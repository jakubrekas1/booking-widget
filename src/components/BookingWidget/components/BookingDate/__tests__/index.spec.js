import React from 'react';
import renderer from 'react-test-renderer';
import BookingDate from '../BookingDateContainer';

describe('Booking date component', () => {
  it('should renders correctly', () => {
    const Component = <BookingDate />;
    const tree = renderer.create(Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
