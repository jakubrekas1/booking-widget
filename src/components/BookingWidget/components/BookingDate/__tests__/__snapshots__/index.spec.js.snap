// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking date component should renders correctly 1`] = `
<div
  className="booking-date__outer"
>
  <div
    className="booking-date"
  >
    <p
      className="booking-date__label"
    >
      Dates
    </p>
    <div
      className="booking-date__wrapper"
    >
      <input
        disabled={true}
        onClick={[Function]}
        placeholder="Check In"
        value=""
      />
      <img
        alt="Arrow"
        className="booking-date__arrow"
        src="../../../../assets/images/arrow-right.svg"
      />
      <input
        disabled={true}
        onClick={[Function]}
        placeholder="Check Out"
        value=""
      />
    </div>
  </div>
</div>
`;
